### PR TITLE
LOOP-1983: Prevent crash in MockService

### PR DESCRIPTION
**Pull request by rickpasetto**
Stack traces seem to indicate a crash in MockService.  Either this array is being illegally accessed from multiple threads (more likely), or it is running out of memory (less likely, but still plausible given that  says he’s set the CGM update every 5 sec and it has run for hours).
[LOOP-1983](https://tidepool.atlassian.net/browse/LOOP-1983)

### CODE REVIEW

File: MockKit/MockService.swift [hunk 1]
```diff
@@ -23,7 +23,10 @@ public final class MockService: Service {
     
     public var analytics: Bool
     
+    public let maxHistoryItems = 1000
+    
     public var history: [String]
+    private let historyQueue = DispatchQueue(label: "MockService_historyQueue")
```
Comment (ps2) [inline MockKit/MockService.swift]: It looks like history is currently being accessed from UI code in MockServiceSettingsViewController.

Putting history writing onto a new thread wouldn't solve the issue, as you'd still have writing from one thread (historyQueue) and reading on another thread (main).

I'd recommend just dispatching the writes onto .main instead. Then both reads and writes will be on .main. No need to create new queue. 

If you did want to just protect this one var, and keep writing on background threads, locking would be preferable to creating a new queue, locking being the more lightweight option.

**Review by ps2 — CHANGES_REQUESTED** [review]

File: MockKit/MockService.swift (hunk 1 above)
Comment (rickpasetto) [reply MockKit/MockService.swift]: Good point.  I see now that `history` is `public`.  I'll fix this to use locking.

File: MockKit/MockService.swift [hunk 2]
```diff
@@ -25,25 +25,24 @@ public final class MockService: Service {
     
     public let maxHistoryItems = 1000
     
-    public var history: [String]
-    private let historyQueue = DispatchQueue(label: "MockService_historyQueue")
+    private var _history = Locked<[String]>([])
```
Comment (ps2) [inline MockKit/MockService.swift]: [nit] the pattern we've been using is to name this variable `lockedHistory`, rather than have an underscore.

**Review by ps2 — APPROVED** [review]
LGTM. Just a naming comment.

**Pull request merged**